source/MaterialXCore/Material.cpp

```diff
@@ -62,28 +62,25 @@ std::unordered_set<NodePtr> getShaderNodes(const NodePtr& materialNode, const st
     for (const InputPtr& input : inputs) 
     {
         // Scan for a node directly connected to the input.
+        // Note that this will handle traversing through interfacename associations.
         //
-        const string& inputShader = input->getNodeName();
-        if (!inputShader.empty())
+        NodePtr shaderNode = input->getConnectedNode();
```

Comment:
Fix for traversing from materials inside nodegraphs to shaders using logic previously added.